src/browser.cc

```diff
@@ -0,0 +1,369 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2015 Dmitry "Dima" Korolev <dmitry.korolev@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#include "helpers.h"
+
+#include "insights.h"
+CEREAL_REGISTER_TYPE(insight::MutualInformation);
+
+#include "html.h"
+
+#include "../../Current/Bricks/dflags/dflags.h"
+#include "../../Current/Bricks/file/file.h"
+#include "../../Current/Bricks/net/api/api.h"
+#include "../../Current/Bricks/strings/util.h"
+#include "../../Current/Bricks/waitable_atomic/waitable_atomic.h"
+
+DEFINE_int32(port, 3000, "Port to spawn the browser on.");
+DEFINE_string(route, "/", "The route to serve the browser on.");
+DEFINE_string(output_uri_prefix, "http://localhost:3000", "The prefix for the URI-s output by the server.");
+
+DEFINE_string(input, "data/insights.json", "Path to the file containing the insights to browse.");
+DEFINE_string(id_key, "you_are_awesome", "The URL parameter name containing smart session token ID.");
+
+using bricks::strings::Printf;
+using bricks::strings::FromString;
+using bricks::strings::ToString;
+using bricks::FileSystem;
+using bricks::WaitableAtomic;
+
+std::string RandomString(const size_t length = 8) {
+  std::string s;
+  for (size_t i = 0; i < length; ++i) {
+    s += 'a' + rand() % 26;
+  }
+  return s;
+}
+
+struct TopLevelResponse {
+  std::string SMART_HTML_BROWSE_URI;
```

Comment:
I am still against using constant-like casing here. There could be something like `smart_html_browse_uri_EXPERIMENTAL` or `smart_html_browse_uri_DEBUG` etc. so that when things settle, you just remove the suffix.